Array helpers all and any behaves weird in else-if

### Describe the bug

Using an array helper function such as `any` in an else-if and referencing the `it` shorthand causes a compiler error.

### Expected Behavior

The code should compile :)

### Current Behavior

> /tmp/v_502/testar.17674342771543642862.tmp.c:2171:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_502/testar.17674342771543642862.tmp.c:12086:13: error: use of undeclared identifier '_t6'
        } else if (_t6) {
                   ^
1 warning and 1 error generated.
builder error: 
==================
C error. This should never happen.

### Reproduction Steps

```vlang
module main

fn main() {
	x := [3, 4]
	y := [2, 3, 4, 5]
	if x.all(it in y) || y.all(it in x) {
		println('all')
	} else if x.any(it in y) {
		println('any')
	}
}
```

### Possible Solution

For now it can be circumvented by placing the second if inside an else like this:

```vlang
module main

fn main() {
	x := [3, 4]
	y := [2, 3, 4, 5]
	if x.all(it in y) || y.all(it in x) {
		println('all')
	} else {
		if x.any(it in y) {
			println('any')
		}
	}
}
```

### Additional Information/Context

_No response_

### V version

0.3.2 fd04c1a

### Environment details (OS name and version, etc.)

macos, macOS, 13.0.1, 22A400

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -948,6 +948,7 @@ fn (mut g Gen) gen_array_any(node ast.CallExpr) {
 	if has_infix_left_var_name {
 		g.indent--
 		g.writeln('}')
+		g.set_current_pos_as_last_stmt_pos()
 	}
 	if s_ends_with_ln {
 		g.writeln(s)

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -1023,6 +1023,7 @@ fn (mut g Gen) gen_array_all(node ast.CallExpr) {
 	if has_infix_left_var_name {
 		g.indent--
 		g.writeln('}')
+		g.set_current_pos_as_last_stmt_pos()
 	}
 	if s_ends_with_ln {
 		g.writeln(s)

--- a/vlib/v/gen/c/if.v
+++ b/vlib/v/gen/c/if.v
@@ -337,12 +337,13 @@ fn (mut g Gen) if_expr(node ast.IfExpr) {
 	}
 	if node.branches.len > 0 {
 		g.writeln('}')
+		g.set_current_pos_as_last_stmt_pos()
 	}
-	g.set_current_pos_as_last_stmt_pos()
 	if needs_tmp_var {
 		if g.infix_left_var_name.len > 0 {
 			g.indent--
 			g.writeln('}')
+			g.set_current_pos_as_last_stmt_pos()
 		}
 		g.empty_line = false
 		g.write('${cur_line} ${tmp}')

--- a/vlib/v/tests/if_expr_with_multi_array_call_test.v
+++ b/vlib/v/tests/if_expr_with_multi_array_call_test.v
@@ -14,3 +14,16 @@ fn foo() string {
 	}
 	return ''
 }
+
+fn test_if_expr_with_several_any_calls() {
+	x := [3, 4]
+	y := [2, 3, 4, 5]
+	if x.any(it in y) || y.any(it in x) {
+		println('then')
+		assert true
+	} else if x.any(it in y) {
+		println('else')
+		assert false
+	}
+	assert true
+}

--- a/vlib/v/tests/if_expr_with_multi_array_call_test.v
+++ b/vlib/v/tests/if_expr_with_multi_array_call_test.v
@@ -1,0 +1,16 @@
+fn test_if_expr_with_multi_array_call() {
+	ret := foo()
+	println(ret)
+	assert ret == 'all'
+}
+
+fn foo() string {
+	x := [3, 4]
+	y := [2, 3, 4, 5]
+	if x.all(it in y) || y.all(it in x) {
+		return 'all'
+	} else if x.any(it in y) {
+		return 'any'
+	}
+	return ''
+}

